File: articles/path-dependence-lock-in-and-decision-timing/python/switching_cost_model.py
#!/usr/bin/env python3
"""Switching cost helper."""

from __future__ import annotations


def switching_cost(investment: float, network_dependence: float, institutional_routine: float) -> float:
    return 0.36 * investment + 0.34 * network_dependence + 0.30 * institutional_routine


if __name__ == "__main__":
    print(round(switching_cost(0.55, 0.62, 0.58), 6))

File: articles/path-dependence-lock-in-and-decision-timing/julia/switching_cost_model.jl
# switching_cost_model.jl

function switching_cost(investment, network_dependence, institutional_routine)
    return 0.36 * investment + 0.34 * network_dependence + 0.30 * institutional_routine
end

println("Switching cost = ", round(switching_cost(0.55, 0.62, 0.58), digits=6))
